### ISSUE TITLE
ES6 Destruct on Catch

### ISSUE THREAD

**Issue by jupl** [eslint/espree#105]
Originally reported on eslint/eslint#2123.

Given the following snippet, with ESLint I get an error on the `catch` line with `Unexpected token {`. There are no errors/warnings reported on the other destructs.

```js
function x({a}) {
  try {
    const {b} = a;
  }
  catch({stack}) {
  }
}
```

**Comment by nzakas** [comment]
Working on this.
